functions/helpers/insightsService.private.ts

```diff
@@ -0,0 +1,89 @@
+import { get } from 'lodash';
+
+type InsightsAttributes = {
+  conversations?: { [key: string]: string | number };
+  customers?: { [key: string]: string | number };
+};
+
+type TaskAttributes = {} & InsightsAttributes;
+
+type InsightsObject = 'customers' | 'conversations';
+
+// Each of this ConfigSpec maps (possibly) many form field to one insights attribute
+type OneToManyConfigSpec = {
+  insightsObject: InsightsObject; // In which attributes object this goes
+  attributeName: string; // Which name the property receives in above object (e.g. customer_attribute_1)
+  questions: string[]; // Array of questions names (as they are named in the bot definition) to grab and concatenate to drop in above property
+};
+type OneToManyConfigSpecs = OneToManyConfigSpec[];
+
+type BotMemory = {
+  memory: {
+    twilio: { collected_data: { collect_survey: { [question: string]: string | number } } };
+  };
+};
+type SurveyInsightsUpdateFunction = (memory: BotMemory) => InsightsAttributes;
+
+const delimiter = ';';
+
+const mergeAttributes = (
+  previousAttributes: TaskAttributes,
+  newAttributes: InsightsAttributes,
+): TaskAttributes => {
+  return {
+    ...previousAttributes,
+    conversations: {
+      ...previousAttributes.conversations,
+      ...newAttributes.conversations,
+    },
+    customers: {
+      ...previousAttributes.customers,
+      ...newAttributes.customers,
+    },
+  };
+};
+
+const applyCustomUpdate = (customUpdate: OneToManyConfigSpec): SurveyInsightsUpdateFunction => {
+  return memory => {
+    const updatePaths = customUpdate.questions.map(
+      question => `twilio.collected_data.collect_survey.${question}.answer`, // Path where the answer for each question should be in bot memory
+    );
+    // concatenate the values, taken from dataSource using paths (e.g. 'contactForm.childInformation.province')
+    const value = updatePaths.map(path => get(memory, path, '')).join(delimiter);
+
+    return {
+      [customUpdate.insightsObject]: {
+        [customUpdate.attributeName]: value,
+      },
+    };
+  };
+};
+
+const bindApplyCustomUpdates = (
+  oneToManyConfigSpecs: OneToManyConfigSpecs,
+): SurveyInsightsUpdateFunction[] => {
+  const customUpdatesFuns = oneToManyConfigSpecs.map(applyCustomUpdate);
+
+  return customUpdatesFuns;
+};
+
+export const buildSurveyInsightsData = (oneToManyConfigSpecs: OneToManyConfigSpecs) => (
+  taskAttributes: TaskAttributes,
+  memory: BotMemory,
+) => {
+  // NOTE: I assume that if surveys are enabled this is not needed, right?
+  // if (!shouldSendInsightsData(task)) return previousAttributes;
```

Comment:
It makes sense that if there's no post survey, it would just return the `previousAttributes`.
(I'm just not sure how `shouldSendInsightsData(task)` works)